src/device.hpp

```diff
@@ -553,6 +553,11 @@ struct cvk_device : public _cl_device_id,
             ->is_bgra_format_not_supported_for_image1d_buffer();
     }
 
+    bool supports_image_format(cl_image_format format) const {
```

Comment:
I think the name is confusing. This function does not return `true` if the format is supported and usable. It returns `true` if the format is not in a deny list. Maybe `is_image_format_disabled` (note that the caller is testing the negative) or better name along those lines?

Also I think we should fold the case currently handled by `is_bgra_format_not_supported_for_image1d_buffer` into this new property query. Fine if you want to do this as a separate PR (we can create an issue for it then).

Reply:
I'll rename this function.
But I don't think `is_bgra_format_not_supported_for_image1d_buffer` should be dealt with inside it.  Those are not at the same level. One is about the CL format, the other is about the Vulkan format.
Everything is taking care of in `cl_image_format_to_vulkan_format` which is the link between the two.

Reply:
Yeah, looking again at how it's used we're doing a bit more than I remembered and remapping the format to another one so we would need a bit more work to merge everything into a single mechanism. Better to wait until we see more cases to attempt to do something.